src/utils/Logger.ts

```diff
@@ -172,6 +172,20 @@ class Logger {
     const logFilePath = this.getReactNativeLogsFilePath()
     console.debug('React Native logs will be piped to ' + logFilePath)
 
+    // If the creation time of the file is more than 28 days ago, delete it.
+    RNFS.stat(logFilePath)
+      .then((stat) => {
+        if (+stat.ctime < +new Date() - 4 * 7 * 24 * 60 * 60 * 1000) {
+          console.debug('Deleting React Native logs file older than 28 days')
+          RNFS.unlink(logFilePath).catch((err) => {
+            console.warn('Failed to delete React Native logs file: ' + err)
+          })
+        }
+      })
+      .catch((err) => {
+        console.warn('Failed get log file stat: ' + err)
+      })
```

Comment:
What do you think of moving this logic to the first `writeLog` call instead and waiting for it before doing the actual write?
Otherwise there's a risk we write the log and delete the file right after. Given these operations are all async and there's no guarantee as which one will be completed first.